ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/HeartbeatAgent.java

```diff
@@ -323,22 +310,23 @@ private boolean decideStatus(final String host, final long windowStart, final lo
 
     @Override
     protected void runOneIteration() {
-      for (Entry<String, HostStatusEntity> hostStatusEntry: hostsStatus.entrySet()) {
-        final String host = hostStatusEntry.getKey();
-        final HostStatusEntity status = hostStatusEntry.getValue();
+      for (Entry<KsqlHost, HostStatus> hostStatusEntry: hostsStatus.get().entrySet()) {
+        final KsqlHost remoteHost = hostStatusEntry.getKey();
         try {
-          if (!host.equals(localHostString)) {
-            final URI remoteUri = buildLocation(localURL, status.getHostInfoEntity().getHost(),
-                                          status.getHostInfoEntity().getPort());
-            LOG.debug("Send heartbeat to host {} at {}", status.getHostInfoEntity().getHost(),
-                      clock.millis());
-            serviceContext.getKsqlClient().makeAsyncHeartbeatRequest(remoteUri, localHostInfo,
-                                                                     clock.millis());
+          if (!remoteHost.equals(localHost)) {
+            final URI remoteUri = ServerUtil.buildRemoteUri(
+                localURL, remoteHost.host(), remoteHost.port());
+            System.out.println(localHost + " Send heartbeat to host " + remoteHost
+                                   + " at timestamp " + System.currentTimeMillis());
+            LOG.debug("Send heartbeat to host {} at {}", remoteHost, clock.millis());
+            serviceContext.getKsqlClient().makeAsyncHeartbeatRequest(
+                remoteUri, localHost, clock.millis());
           }
         } catch (Throwable t) {
-          LOG.error("Request to server: " + status.getHostInfoEntity().getHost() + ":"
-                        + status.getHostInfoEntity().getPort()
-                        + " failed with exception: " + t.getMessage(), t);
+          System.out.println(localHost + " FAILED to Send heartbeat to host " + remoteHost
+                                 + " at timestamp " + System.currentTimeMillis());
```

Comment:
ditto - please make sure to do a find-and-replace throughout the changed files to see if any others slipped through